### PR TITLE
Introduce unit tests

### PR DESCRIPTION
**Pull request by SiskaPavel**

### CODE REVIEW

File: cmake/googletest.cmake [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#
+# This CMake script integrates Google Test into the project, allowing easy usage.
+# Usage example: target_link_libraries(my_target PRIVATE GTest::GTest)
+
+include(FetchContent)
+
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG        release-1.12.1
```
Comment (Lukas955) [inline cmake/googletest.cmake:13]: It there any reason not to use current version `v1.14.0`?

File: cmake/googletest.cmake [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+)
+FetchContent_MakeAvailable(googletest)
+add_library(GTest::GTest INTERFACE IMPORTED)
+target_link_libraries(GTest::GTest INTERFACE gtest_main)
```
Comment (Lukas955) [inline cmake/googletest.cmake:17]: I believe that these two commands (add_library & target_link_libraries) are not required. All you need to do, is to link against `GTest::gtest_main` in the next commit.

File: src/telemetry/CMakeLists.txt [hunk 3]
```diff
@@ -26,3 +26,10 @@ if (TELEMETRY_INSTALL_TARGETS)
 	install(TARGETS telemetry LIBRARY DESTINATION ${INSTALL_DIR_LIB})
 	install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION ${INSTALL_DIR_INCLUDE})
 endif()
+
+if (TELEMETRY_ENABLE_TESTS)
+	add_executable(testTelemetry ${TELEMETRY_SOURCE_FILES})
+	target_compile_definitions(testTelemetry PRIVATE TELEMETRY_ENABLE_TESTS)
+	target_link_libraries(testTelemetry telemetry GTest::GTest)
+	add_test(testTelemetry testTelemetry)
```
Comment (Lukas955) [inline src/telemetry/CMakeLists.txt:34]: Replace `add_test` with following for better integration with CTest:
```
include(GoogleTest)
gtest_discover_tests(testTelemetry)
```

File: cmake/googletest.cmake (hunk 1 above)
Comment (SiskaPavel) [reply cmake/googletest.cmake]: Updated to newer version

File: cmake/googletest.cmake (hunk 2 above)
Comment (SiskaPavel) [reply cmake/googletest.cmake]: Updated

File: src/telemetry/CMakeLists.txt (hunk 3 above)
Comment (SiskaPavel) [reply src/telemetry/CMakeLists.txt]: Updated

File: CMakeLists.txt [hunk 4]
```diff
@@ -32,8 +33,16 @@ endif()
 
 include(cmake/dependencies.cmake)
 
+if (TELEMETRY_ENABLE_TESTS)
+	include(cmake/googletest.cmake)
+	include(GoogleTest)
+    enable_testing()
```
Comment (Lukas955) [inline CMakeLists.txt:39]: Different indentations. The same also applies to the cmake/googletest.cmake file.

File: src/telemetry/CMakeLists.txt [hunk 5]
```diff
@@ -26,3 +26,10 @@ if (TELEMETRY_INSTALL_TARGETS)
 	install(TARGETS telemetry LIBRARY DESTINATION ${INSTALL_DIR_LIB})
 	install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION ${INSTALL_DIR_INCLUDE})
 endif()
+
+if (TELEMETRY_ENABLE_TESTS)
+	add_executable(testTelemetry ${TELEMETRY_SOURCE_FILES})
+	target_compile_definitions(testTelemetry PRIVATE TELEMETRY_ENABLE_TESTS)
+	target_link_libraries(testTelemetry telemetry GTest::gtest_main)
```
Comment (Lukas955) [inline src/telemetry/CMakeLists.txt:33]: Is it really necessary to link it against `telemetry` library as we build independent binary that contains all functions?

**Review by Lukas955 — APPROVED** [review]

**Pull request merged**